server/chat-commands/room-settings.ts

```diff
@@ -821,6 +822,8 @@ export const commands: ChatCommands = {
 			return this.errorReply(`An error occured while renaming the room.`);
 		}
 		this.modlog(`RENAMEROOM`, null, `from ${oldTitle}`);
+		room.log.roomlogFilename = "";
+		await room.log.setupRoomlogStream(true);
```

Comment:
Actually, in hindsight, this should also be unnecessary since `room.log.rename` (which is called by `room.rename`) should already handle it.

```suggestion
```

Is there a problem with its current implication?

Reply:
Tested this myself, and it seems to function fine currently, other than the fact you can ``>> room.rename()`` and it doesn't fail (which it probably should.)
I may be missing something, though.

Reply:
``room.log.rename`` and ``room.rename`` should both be throwing errors if there's no ID and title given.

Reply:
it does not log any messages in the chat after renaming it which is why i put the ``await room.log.setupRoomlogStream(true);`` which not necessarily a big issue but could be also be an issue if say a room was renamed and there hadn't been a restart and staff needed to see the logs with ``/chatlog`` but found none since the rename

Reply:
https://github.com/smogon/pokemon-showdown/blob/master/server/rooms.ts#L1379
https://github.com/smogon/pokemon-showdown/blob/master/server/roomlogs.ts#L247
@mia-pi-git seems no one bothered to add something to make it throw errors if there's no IDs given so that's fun lol